tensorflow_probability/python/experimental/mcmc/rhat_reducer.py

```diff
@@ -0,0 +1,203 @@
+# Copyright 2020 The TensorFlow Probability Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ============================================================================
+"""RhatReducer for calculating streaming potential scale reduction."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+
+# Dependency imports
+import tensorflow.compat.v2 as tf
+
+from tensorflow_probability.python.experimental.mcmc import reducer as reducer_base
+from tensorflow_probability.python.experimental.stats import sample_stats
+from tensorflow_probability.python.internal import prefer_static as ps
+from tensorflow_probability.python.mcmc.internal import util as mcmc_util
+
+
+__all__ = [
+    'RhatReducer',
+]
+
+
+RhatReducerState = collections.namedtuple(
+    'RhatReducerState', 'sample_shape, rhat_state')
+
+
+class RhatReducer(reducer_base.Reducer):
+  """`Reducer` that computes a running rhat diagnostic statistic.
+
+  `RhatReducer` assumes that all provided chain states include samples from
+  multiple independent Markov chains, and that all of these chains are to
+  be included in the same calculation. `RhatReducer` also assumes that
+  incoming samples have shape `[Ni, Ci1, Ci2,...,CiD]`. Dimension `0` indexes
```

Comment:
Why not also allow an arbitrary "event shape", as in `tfp.mcmc.potential_scale_reduction_factor`?

Reply:
Pretty sure that's supported but I didn't explicitly state it in the Docstring... oops